### PR TITLE
Add command events to implement maneuver to pitch and roll about sun line

### LINKED ISSUE #316: NSM offset command event for the Command Events sheet

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This adds two new command events for the Chandra Comment Events sheet:
- `Maneuver sun pitch`: maneuver to a specified absolute sun pitch angle
- `Maneuver sun rasl`: roll about sun line by specified angle

Note that this passes tests with or without https://github.com/sot/ska_sun/pull/37 which changes the sign convention of roll about the sun line. This is the case because the code applies the rasl using `apply_sun_pitch_yaw` and then measures the change using `get_sun_pitch_yaw`. Those use the same sign convention so `rasl2 - rasl1` is the same in both cases (with/without 37).

Closes #316

## Interface impacts

No code impacts but this adds new command events. When this is ready to merge I will update the FOT MP command events TWiki documentation to include the two new command event types.

### Unit tests

Tested with https://github.com/sot/ska_sun/pull/37 in the PYTHONPATH.

- [x] Mac
```
(ska3) ➜  kadi git:(nsm-offset-pitch) printenv PYTHONPATH                                   
/Users/aldcroft/git/ska_sun
(ska3) ➜  kadi git:(nsm-offset-pitch) python -c "import ska_sun; print(ska_sun.__version__)"
3.13.1.dev1+g3579ca1
(ska3) ➜  kadi git:(nsm-offset-pitch) git rev-parse HEAD                                    
31358c88273b96c9ece0899c7320048303aa2faf
(ska3) ➜  kadi git:(nsm-offset-pitch) pytest
======================================================== test session starts ========================================================
platform darwin -- Python 3.10.8, pytest-7.2.1, pluggy-1.0.0
rootdir: /Users/aldcroft/git, configfile: pytest.ini
plugins: timeout-2.1.0, anyio-3.6.2
collected 233 items                                                                                                                 

kadi/commands/tests/test_commands.py .......................................................................................  [ 37%]
kadi/commands/tests/test_states.py .......................x..............................................x................... [ 75%]
....                                                                                                                          [ 77%]
kadi/commands/tests/test_validate.py ....................                                                                     [ 86%]
kadi/tests/test_events.py ..........                                                                                          [ 90%]
kadi/tests/test_occweb.py ......................                                                                              [100%]

```

Independent check of unit tests by Jean with a ska3-masters with ska_sun PR up-to-date
- [x] OSX
```

(ska3) flame:kadi jean$ pytest
===================================================================== test session starts =====================================================================
platform darwin -- Python 3.10.8, pytest-7.2.1, pluggy-1.0.0
rootdir: /Users/jean/git, configfile: pytest.ini
plugins: timeout-2.1.0, anyio-3.6.2
collected 233 items                                                                                                                                           

kadi/commands/tests/test_commands.py .......s.......s......................................................................s                            [ 37%]
kadi/commands/tests/test_states.py .......................x..............................................x.......................                       [ 77%]
kadi/commands/tests/test_validate.py ....................                                                                                               [ 86%]
kadi/tests/test_events.py ..........                                                                                                                    [ 90%]
kadi/tests/test_occweb.py ......................                                                                                                        [100%]

====================================================================== warnings summary =======================================================================
kadi/kadi/commands/tests/test_commands.py::test_get_starcats_each_year[year0]
  /Users/jean/miniconda3/envs/ska3/lib/python3.10/site-packages/setuptools_scm/git.py:295: UserWarning: git archive did not support describe output
    warnings.warn("git archive did not support describe output")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================== 228 passed, 3 skipped, 2 xfailed, 1 warning in 91.89s (0:01:31) ===============================================
(ska3) flame:kadi jean$ git rev-parse HEAD
31358c88273b96c9ece0899c7320048303aa2faf
(ska3) flame:kadi jean$ python -c "import ska_sun; print(ska_sun.__version__)"
3.13.1.dev2+g2ec26c1
```

### Functional tests

No functional testing.

### CODE REVIEW

File: kadi/commands/states.py [hunk 1]
```diff
@@ -1394,6 +1394,93 @@ def callback(cls, date, transitions, state, idx):
...
+
+        # If current attitude is not defined then just drop the NSM maneuver on
+        # the floor. The state will start getting defined when the first normal
+        # maneuver happens.
+        if None in curr_att:
+            return
+
+        curr_att = Quat(curr_att)
+        sun_ra, sun_dec = ska_sun.position(date)
+        targ_att = ska_sun.apply_sun_pitch_yaw(
```
Comment (jeanconn) [inline kadi/commands/states.py:1474]: I'll come back to review this when we've decided on the sign of apply_sun_pitch_yaw.

Comment (taldcroft) [reply kadi/commands/states.py:1474]: The sign convention is now defined with https://github.com/sot/ska_sun/pull/37.

**Comment by jeanconn** [comment]
Per our conversation, it would be good to include something short in the twiki docs about how the common use case for rasl would be a yaw change, and that under the hood it is implemented as a maneuver.

**Comment by taldcroft** [comment]
> Per our conversation, it would be good to include something short in the twiki docs about how the common use case for rasl would be a yaw change, and that under the hood it is implemented as a maneuver.

Done.

**Pull request merged**